### PR TITLE
Allow for marketplace provided plugin cover images

### PR DESCRIPTION
**Pull request by michalkleiner**
### Description:

The change in this PR allows for use of cover images provided by the plugins marketplace while keeping the existing fallback images for Matomo vs 3rd party plugins.

The PR:

- adds a mechanism to display a cover image specified in the plugin's metadata
- tweaks the existing fallback logic so that all paid Matomo plugins without a cover image use the blue Matomo image and all other free plugins regardless of their vendor use the generic grey plugin image
- adds a mechanism via the test config to replace remote image URLs with local test image URLs for UI tests
- updates file formatting for all static marketplace fixtures for better legibility

There is also a follow-up draft PR #21915 which should be merged once the corresponding changes to the marketplace have been deployed so that the test there can pass.

ref. DEV-17378

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Marketplace/templates/plugin-list.twig [hunk 1]
```diff
@@ -8,7 +8,7 @@
                             {% import '@Marketplace/macros.twig' as marketplaceMacro %}
                             {% import '@CorePluginsAdmin/macros.twig' as pluginsMacro %}
 
-                                <img src="{{ plugin.previewImage }}" alt="" class="preview-image">
+                                <img src="{{ plugin.coverImage }}?w=880&h=480" alt="" class="cover-image">
```
Comment (michalkleiner) [inline plugins/Marketplace/templates/plugin-list.twig:11]: The query params specifying expected width and height of the image from marketplace do no harm if the image is local, so no extra logic was applied here.

**Comment by michalkleiner** [comment]
Failure for `Piwik\Plugins\Marketplace\tests\System\Api\ClientTest::test_getPluginInfo_existingPluginOnTheMarketplace` within the SystemTestsPlugins suites (for all 3 PHP versions) is expected until a corresponding change providing the `category` and `coverImage` fields is deployed to the marketplace.

I'd be happy to remove the change to the test and add it back after the marketplace change is deployed to production if that's preferred.

**UPDATE**: Discussed in person with MC and will move this to a separate PR that can be merged once the corresponding change is deployed to Marketplace PROD.

**UPDATE 2**: Moved to PR #21915

**Comment by michalkleiner** [comment]
This change can be reviewed, merged and released independently of the marketplace changes as it provides defaults for the category and cover image in the same way how it worked before.

File: plugins/Marketplace/tests/resources/v2.0_consumer-access_token-consumer1_paid2_custom1.json
Comment (michalkleiner) [inline plugins/Marketplace/tests/resources/v2.0_consumer-access_token-consumer1_paid2_custom1.json:1]: All the plugin's mock payloads have been reformatted for better legibility in a separate commit.

File: plugins/Marketplace/config/test.php [hunk 2]
```diff
@@ -83,11 +83,37 @@
 
         $service->authenticate($accessToken);
 
-        function removeReviewsUrl($content)
+        function updatePluginUrlsForTests(&$plugin)
```
Comment (caddoo) [inline plugins/Marketplace/config/test.php:87]: It's quite hard to understand what this code is doing, would it be possible to name the function with something that better represents what its doing?

File: plugins/Marketplace/config/test.php [hunk 3]
```diff
@@ -83,11 +83,37 @@
...
+                        'plugins/Marketplace/tests/resources/images/plugins/$1/images/$2/$3',
+                        'plugins/Marketplace/tests/resources/images/categories/$1',
+                    ],
+                    $plugin['coverImage'],
+                    1
+                );
+            }
+        }
+
+        function updatePayload($content)
```
Comment (caddoo) [inline plugins/Marketplace/config/test.php]: Also update the name of this, so its intent is clear and you don't have to read the code to understand.

File: plugins/Marketplace/config/test.php (hunk 2 above)
Comment (michalkleiner) [reply plugins/Marketplace/config/test.php:87]: Yep, I can try and find a better name for it. Or we can keep the original method updating one aspect of the payload and create another one with more descriptive name also updating the payload and then call both of them several times on the test payloads.

I sort of saw this as inferior to the actual functional code changes that affect the outcome in the UI. This was really only adjusted to support providing local mock images for UI tests.

Comment (caddoo) [reply plugins/Marketplace/config/test.php:87]: Whatever you think makes sense here so that the person coming to look at this code next isn't confused.

Comment (michalkleiner) [reply plugins/Marketplace/config/test.php:87]: Updated method names and added inline comments to better explain what they do.

File: plugins/Marketplace/config/test.php (hunk 3 above)
Comment (michalkleiner) [reply plugins/Marketplace/config/test.php]: Updated method names and added inline comments to better explain what they do.

**Review by sgiehl — APPROVED** [review]
Looks good to me overall. Not sure how much sense it makes to merge it before the Marketplace change was applied.

**Comment by michalkleiner** [comment]
Thanks @sgiehl. I know we don't want dead code that won't be used though I'm confident this will get through at the Marketplace end as well. Also, the change surface here is quite contained, so I think this can be merged. If we need to adjust anything I can follow up on that.

**Pull request merged**